### ISSUE TITLE
Reduce generated permissions to `read-all` at top level for generated workflow file

### ISSUE THREAD

**Issue by andrewvaughan** [oxsecurity/megalinter#3026]
**Describe the bug**
Checkov (rightfully) complains about `CKV2_GHA_1` that the workflow file leaves permissions to the default `write-all` in `.mega-linter.yml` file. This line should be added to the top level:

```yml
permissions: read-all
```

The one build step permissions shouldn't have to change. It's a slightly pedantic issue, but it does occur on the default setup. It will also help prevent any mistakes if users add a second step beyond `build`.

**Comment by nvuillam** [comment]
@andrewvaughan to post comments with results or push new commits, read-all won't be enough :)

But MegaLinter users are free to not use these features and set manually permissions to read-all :)

**Comment by andrewvaughan** [comment]
Correct! But you should be able to put these permissions per-job.  The `github-actions` linter is complaining because `permissions: read-only` is not set at the root.  This requires each job to be explicit in setting its permissions.

So to be clear, this isn't replacing the existing `permissions` block, this is adding an additional `permissions: read-only` above the `on` clause to set `read-only` as a build default. The existing `permission` block will overwrite this.

As mentioned, it's pedantic, but it does enforce any extra build steps to require explicit permission building per-job, which is best practice. That is what Checkov errors about in the existing workflow.

**Comment by nvuillam** [comment]
@andrewvaughan i fully agree with your clear explanations and will of course accept an incoming PR :)

**Comment by andrewvaughan** [comment]
> @andrewvaughan i fully agree with your clear explanations and will of course accept an incoming PR :)

I got you fam.
